### ISSUE TITLE
Data lost when docker killed and downloaded to buffer

### ISSUE THREAD

**Issue by piter-pit** [transmission/transmission#7283]
### What is the issue?

I have quite high buffer set on my docker transmission, and noted that each time when I update docker (the container is killed) or I have power lost - data downloaded to the buffer (but not saved to disk) are marked as downloaded, so after 100% download torrent is marked as fully downloaded, but some data is corrupted. It is not visible until you run verify local data that shows some parts are missing.
The solution might be to mark part as downloaded after it was written to disk, or add possibility to verify local data after download, or after download when interruption was detected.
I believe it's not docker version issue but generic

### Which application of Transmission?

transmission-daemon

**Comment by killemov** [comment]
> I believe it's not docker version issue but generic

Correct. The same thing happens when the disk is full. The pieces that could not have been written to disk are marked as downloaded, but will be noticed as missing or corrupt only after verifying. And then sometimes those pieces will not be downloaded again until the torrent is recycled or Transmission is restarted again.

This is another variant of #4724
